packages/@apphosting/adapter-nextjs/src/bin/build.ts

```diff
@@ -2,12 +2,69 @@
 import { spawnSync } from "child_process";
 import { loadConfig, readRoutesManifest } from "../utils.js";
 
+import { join } from "path";
+import fsExtra from "fs-extra";
+import { stringify as yamlStringify } from "yaml";
+
+// unable to use shorthand imports on fsExtra since fsExtra is CJS
+const { move, exists, writeFile, mkdirp } = fsExtra;
 const cwd = process.cwd();
 
-spawnSync("next", ["build"], { cwd, stdio: "inherit", env: { ...process.env, NEXT_PRIVATE_STANDALONE: 'true' } });
+// Set standalone mode
+process.env.NEXT_PRIVATE_STANDALONE = "true"; 
+// Opt-out sending telemetry to Vercel
+process.env.NEXT_TELEMETRY_DISABLED = "1"; 
+
+build(cwd);
+
+const {distDir} = await loadConfig(cwd);
+const manifest = await readRoutesManifest(join(cwd, distDir));
+
+const appHostingOutputDirectory = join(cwd, ".apphosting");
+const appHostingStaticDirectory = join(appHostingOutputDirectory,"_next", "static");
+const appHostingPublicDirectory = join(appHostingOutputDirectory, "public");
+const outputBundlePath = join(appHostingOutputDirectory, "bundle.yaml");
+const serverFilePath = join(appHostingOutputDirectory, "server.js");
+
+const standaloneDirectory = join(cwd, distDir, "standalone");
+const staticDirectory = join(cwd, distDir, "static");
+const publicDirectory = join(cwd, "public");
+
+await mkdirp(appHostingStaticDirectory);
+
+// Run build command
+function build(cwd: string) {
+    spawnSync("npm run", ["build"], {cwd, shell: true, stdio: "inherit"}); 
```

Comment:
I think the syntax here might be wrong I think probably should be 
spawnSync("npm", ["run", "build"], {cwd, shell: true, stdio: "inherit"});
 instead